gulpfile.js

```diff
@@ -29,22 +34,125 @@ var tsCompileOptions = {
     "removeComments": true
 };
 
-function executeCommand(command, args, callback) {
-    var process = child_process.exec(command + " " + args.join(" "));
+function spawnCommand(command, args, callback, silent, detached) {
+    var options = {};
+    if (detached) {
+        options.detached = true;
+        options.stdio = ["ignore"];
+    }
+    
+    var process = child_process.spawn(command, args, options);
+
+    process.stdout.on('data', function (data) {
+        if (!silent) console.log("" + data);
+    });
 
-    process.stdout.on('data', function(data) {
-        console.log("" + data);        
+    process.stderr.on('data', function (data) {
+        if (!silent) console.error("" + data);
     });
+
+    if (!detached) {
+        process.on('exit', function (code) {
+            callback && callback(code === 0 ? undefined : "Error code: " + code);
+        });
+    }
     
-    process.stderr.on('data', function(data) {
-        console.error("" + data);
+    return process;
+};
+
+function execCommand(command, args, callback, silent) {
+    var process = child_process.exec(command + " " + args.join(" "));
+
+    process.stdout.on('data', function (data) {
+        if (!silent) console.log("" + data);
     });
+
+    process.stderr.on('data', function (data) {
+        if (!silent) console.error("" + data);
+    });
+    
+    process.on('error', function (code) {
```

Comment:
[The callback arg should be an `error` object.](https://nodejs.org/api/child_process.html#child_process_event_error)